GitUI/GitUICommands.cs

```diff
@@ -524,48 +526,64 @@ public bool StartCommitDialog(IWin32Window? owner, string? commitMessage = null,
                 return false;
             }
 
+            // Commit dialog can be opened on its own without the main form
+            // If it is opened by itself, we need to ensure plugins are loaded because some of them
+            // may have hooks into the commit flow
+            bool werePluginsRegistered = PluginRegistry.PluginsRegistered;
+
+            try
+            {
+                // Load plugins synchronously
+                // if the commit dialog is opened from the main form, all plugins are already loaded and we return instantly,
+                // if the dialog is loaded on its own, plugins need to be loaded before we load the form
+                if (!werePluginsRegistered)
+                {
+                    PluginRegistry.InitializeForCommitForm();
+                    PluginRegistry.Register(this);
+                }
+            }
+            catch (Exception exception)
+            {
+                // Nothing: we don't want plugin loading to crash the application here
+                Trace.WriteLine(exception);
+            }
+
             bool Action()
             {
                 ThreadHelper.ThrowIfNotOnUIThread();
 
-                // Commit dialog can be opened on its own without the main form
-                // If it is opened by itself, we need to ensure plugins are loaded because some of them
-                // may have hooks into the commit flow
-                bool werePluginsRegistered = PluginRegistry.PluginsRegistered;
+                using FormCommit form = new(this, commitMessage: commitMessage);
+                if (showOnlyWhenChanges)
+                {
+                    form.ShowDialogWhenChanges(owner);
+                }
+                else
+                {
+                    form.ShowDialog(owner);
+                }
 
+                return true;
+            }
+
+            try
+            {
+                return DoActionOnRepo(owner, Action, changesRepo: false, preEvent: PreCommit, postEvent: PostCommit);
+            }
+            finally
+            {
                 try
                 {
-                    // Load plugins synchronously
-                    // if the commit dialog is opened from the main form, all plugins are already loaded and we return instantly,
-                    // if the dialog is loaded on its own, plugins need to be loaded before we load the form
                     if (!werePluginsRegistered)
```

Comment:
?
```suggestion
                    if (werePluginsRegistered)
```

Reply:
No, it's OK like that. If you come from FormBrowse, the plugins have been already loaded and registered and you have nothing to do (`PluginRegistry.PluginsRegistered` is true).
If you open the FormCommit directly from the File Explorer, no plugins are registered and you want to load and register them (to be exact, only the ones exporting `IGitPluginForCommit` so that it is not slow)